src/client/actions/airdcpp.actions.tsx

```diff
@@ -32,180 +37,206 @@ function sleep(ms: number): Promise<NodeJS.Timeout> {
+    switch (status) {
+      case "connected":
         dispatch({
-          type: AIRDCPP_SEARCH_IN_PROGRESS,
+          type: AIRDCPP_SOCKET_CONNECTED,
+          data: payload,
         });
+        break;
 
-        // We want to get notified about every new result in order to make the user experience better
-        await ADCPPSocket.addListener(
-          `search`,
-          "search_result_added",
-          async (groupedResult) => {
-            // ...add the received result in the UI
-            // (it's probably a good idea to have some kind of throttling for the UI updates as there can be thousands of results)
-
-            dispatch({
-              type: AIRDCPP_SEARCH_RESULTS_ADDED,
-              groupedResult,
-            });
-          },
-          instance.id,
-        );
-
-        // We also want to update the existing items in our list when new hits arrive for the previously listed files/directories
-        await ADCPPSocket.addListener(
-          `search`,
-          "search_result_updated",
-          async (groupedResult) => {
-            // ...update properties of the existing result in the UI
-            dispatch({
-              type: AIRDCPP_SEARCH_RESULTS_UPDATED,
-              groupedResult,
-            });
-          },
-          instance.id,
-        );
+      case "disconnected":
+        dispatch({
+          type: AIRDCPP_SOCKET_DISCONNECTED,
+          data: payload,
+        });
+        break;
 
-        // We need to show something to the user in case the search won't yield any results so that he won't be waiting forever)
-        // Wait for 5 seconds for any results to arrive after the searches were sent to the hubs
-        await ADCPPSocket.addListener(
-          `search`,
-          "search_hub_searches_sent",
-          async (searchInfo) => {
-            await sleep(5000);
-
-            // Check the number of received results (in real use cases we should know that even without calling the API)
-            const currentInstance = await ADCPPSocket.get(
-              `search/${instance.id}`,
-            );
-            if (currentInstance.result_count === 0) {
-              // ...nothing was received, show an informative message to the user
-              console.log("No more search results.");
-            }
-
-            // The search can now be considered to be "complete"
-            // If there's an "in progress" indicator in the UI, that could also be disabled here
-            dispatch({
-              type: AIRDCPP_HUB_SEARCHES_SENT,
-              searchInfo,
-              instance,
-            });
-          },
-          instance.id,
+      default:
+        console.log("Can't set AirDC++ socket status.");
+        break;
+    }
+  };
+export const search =
+  (data: SearchData, ADCPPSocket: any, credentials: any) =>
+  async (dispatch) => {
+    try {
+      if (!ADCPPSocket.isConnected()) {
+        await ADCPPSocket.connect(
+          credentials.username,
```

Comment:
Shouldn't you just call `connect` without any arguments since everything is set in the socket options?

Reply:
Yeah, I never tried testing it without params, after I implemented the change you suggested @maksis. I'll remove the params and push.

Reply:
@maksis Addressed in https://github.com/rishighan/threetwo/pull/58/commits/3102b489c8db55594a79ba84a9789dc9cd65493f

Reply:
> @maksis Addressed in [3102b48](https://github.com/rishighan/threetwo/commit/3102b489c8db55594a79ba84a9789dc9cd65493f)

But the same code is still everywhere...

Reply:
Grrrr... jk. Yes I'll update that

Reply:
Aaaand done: https://github.com/rishighan/threetwo/pull/58/commits/1d8dd3ca3e6d9b0b4a3cb4f6ba462a3de19daacf